HP-79 LocaleContextResolver 등록

jiheon2234 opened the pull request:
# 티켓
HP-79

# 설명
리펙토링하기엔 코드가 너무 많아져서
LocaleContextResolver 등록/테스트까지만 진행

각각 Locale, timeZone 정보 받게 등록
CustomHeaders 스태틱 클래스에서 헤더정보 받게 변경
언어정보에 따라 Content-language 헤더 응답하게 변경

## 타입
- [ ] 버그
- [ ] 작업
- [ ] 기능 향상

# 테스트 방법
코드와 관련하여 어떻게 테스트하였는지 작성해주세요

# 체크리스트
- [x] 작성한 코드가 새로운 에러를 만들지 않았습니다
- [x] 코드가 코드 컨벤션에 모두 만족하는지 확인하였습니다
- [x] 작성한 코드에 관한 unit test를 작성하였고 모든 테스트를 통과하였습니다

Commit:
HP-79
각각 Locale, timeZone 정보 받게 등록
CustomHeaders 스태틱 클래스에서 헤더정보 받게 변경
언어정보에 따라 Content-language 헤더 응답하게 변경

DongHun3946 commented on src/main/java/com/happypill/api/config/localeContextResolver/HappypillLocaleContextResolver.java:
```diff
@@ -0,0 +1,32 @@
+    @Override
+    public void setLocaleContext(HttpServletRequest request, HttpServletResponse response, LocaleContext localeContext) {
+        throw new AssertionError("unreachable");
+    }
```
이 부분은 구현 안하면 요청마다 언어를 변경해줘야하나요? 
![image](https://github.com/user-attachments/assets/548f2829-0bc4-4373-af34-3f8d770eeeef)
예를 들면 위 이미지에서 상품들을 영어로 보려면 페이지를 이동할 때마다 언어를 변경해줘야 하는 지 궁금합니다

jiheon2234 replied:
저희 서버는 유저의 상태를 저장하지 않아요
이동할때마다 프론트에서 언어 파라미터를 넘겨야합니다

DongHun3946 replied:
넵 확인했습니다!

DongHun3946 commented on src/main/java/com/happypill/api/config/CustomHeaders.java:
```diff
@@ -0,0 +1,13 @@
+
+    public static final String LANGUAGE_HEADER = "APP_LANGUAGE";
+
+    public static final String TIMEZONE_HEADER = "APP-TIME-ZONE";
```
헤더 이름을 _ 나 - 로 둘 중 하나로 통일하는 건 어떤가요?

Commit:
HP-79
각각 Locale, timeZone 정보 받게 등록
CustomHeaders 스태틱 클래스에서 헤더정보 받게 변경
언어정보에 따라 Content-language 헤더 응답하게 변경

DongHun3946 reviewed (approved):
고생하셨습니다
